pkg/compare/compare_test.go

```diff
@@ -662,3 +662,230 @@ metadata:
 		})
 	}
 }
+
+func TestYAMLCmpWithYamlInline(t *testing.T) {
+	tests := []struct {
+		desc string
+		a    string
+		b    string
+		want interface{}
+	}{
+		{
+			desc: "ConfigMap data identical",
+			a: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    apiVersion: admissionregistration.k8s.io/v1beta1
+    kind: ValidatingWebhookConfiguration
+    metadata:
+      name: istio-galley`,
+			b: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    apiVersion: admissionregistration.k8s.io/v1beta1
+    kind: ValidatingWebhookConfiguration
+    metadata:
+      name: istio-galley`,
+			want: ``,
+		},
+		{
+			desc: "ConfigMap data order changed",
+			a: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    kind: ValidatingWebhookConfiguration
+    apiVersion: admissionregistration.k8s.io/v1beta1
+    metadata:
+      name: istio-galley
+      value: foo`,
+			b: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    apiVersion: admissionregistration.k8s.io/v1beta1
+    kind: ValidatingWebhookConfiguration
+    metadata:
+      value: foo
+      name: istio-galley`,
+			want: ``,
+		},
+		{
+			desc: "ConfigMap data value change",
+			a: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    apiVersion: admissionregistration.k8s.io/v1beta1
+    kind: ValidatingWebhookConfiguration
+    metadata:
+      name: istio-galley`,
+			b: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    apiVersion: admissionregistration.k8s.io/v1beta2
+    kind: ValidatingWebhookConfiguration
+    metadata:
+      name: istio-galley`,
+			want: `data:
+  validatingwebhookconfiguration.yaml:
+    apiVersion: admissionregistration.k8s.io/v1beta1 -> admissionregistration.k8s.io/v1beta2
+`,
+		},
+		{
+			desc: "ConfigMap data string change",
```

Comment:
suggest omitting this test since the one below is a superset. this will reduce the amount of tests to update if anything changes.

Reply:
Changed test description. This is intended as a deep nested change test.

Reply:
Also simplified and improved the test input.